README.md

```diff
@@ -48,96 +48,120 @@ your Cloud SQL instance.
+configuration.
 
-``` go
+```go
 import (
-	"context"
-	"net"
+    "database/sql"
 
-	"cloud.google.com/go/cloudsqlconn"
-	"github.com/jackc/pgx/v4/pgxpool"
+    "cloud.google.com/go/cloudsqlconn"
+    "cloud.google.com/go/cloudsqlconn/mysql/mysql"
 )
 
 func connect() {
-	// Configure the driver to connect to the database
-	dsn := "user=myuser password=mypass dbname=mydb sslmode=disable"
-	config, err := pgxpool.ParseConfig(dsn)
-	if err != nil {
-		/* handle error */
-	}
+    cleanup, err := mysql.RegisterDriver("cloudsql-mysql", cloudsqlconn.WithCredentialsFile("key.json"))
+    if err != nil {
+        // ... handle error
+    }
+    defer cleanup()
 
-	// Create a new dialer with any options
-	d, err := cloudsqlconn.NewDialer(context.Background())
-	if err != nil {
-		/* handle error */
-	}
-	defer d.Close()
+    db, err := sql.Open(
+        "cloudsql-mysql",
+        "myuser:mypass@cloudsql-mysql(my-project:us-central1:my-instance)/mydb",
+	)
+    // ... etc
+}
+```
 
-	// Tell the driver to use the Cloud SQL Go Connector to create connections
-	config.ConnConfig.DialFunc = func(ctx context.Context, _ string, instance string) (net.Conn, error) {
-		return d.Dial(ctx, "project:region:instance")
-	}
+### SQL Server
 
-	// Interact with the dirver directly as you normally would
-	conn, err := pgxpool.ConnectConfig(context.Background(), config)
-	if err != nil {
-		/* handle error */
-	}
+To use `database/sql`, use `mssql.RegisterDriver` with any necessary Dialer
+configuration.
+
+``` go
+import (
+    "database/sql"
+
+    "cloud.google.com/go/cloudsqlconn"
+    "cloud.google.com/go/cloudsqlconn/sqlserver/mssql"
+)
+
+func connect() {
+    cleanup, err := mssql.RegisterDriver("cloudsql-sqlserver", cloudsqlconn.WithCredentialsFile("key.json"))
+    if err != nil {
+        // ... handle error
+    }
+    defer cleanup()
+
+    db, err := sql.Open(
+        "cloudsql-sqlserver",
+        "sqlserver://user:password@localhost?database=mydb&cloudsql=my-proj:us-central1:my-inst",
+    )
     // ... etc
 }
 ```
 
-[dial-func]: https://pkg.go.dev/github.com/jackc/pgconn#Config
+### APIs and Services
+
+This package requires the following to successfully make Cloud SQL Connections:
 
-### MySQL
+- IAM principal (user, service account, etc.) with the
+[Cloud SQL Client][client-role] role or equivalent. This IAM principal will
+ be used for [credentials](#credentials).
+- The [Cloud SQL Admin API][admin-api] to be enabled within your Google Cloud
+Project. By default, the API will be called in the project associated with
+the IAM principal.
 
-The [Go MySQL driver][mysql] does not provide a stand-alone interface for
-interacting with a database and instead uses `database/sql`. See [the section
-below](#MySQL) on how to use the `database/sql` package with a Cloud SQL MySQL
-instance.
+[admin-api]: https://console.cloud.google.com/apis/api/sqladmin.googleapis.com
+[client-role]: https://cloud.google.com/sql/docs/mysql/roles-and-permissions
```

Comment:
not totally sure how I feel about the flow of having this section being after the code samples. This is basically a pre-req section but we are now moving it below the application code?

Reply:
Agreed. I'll revert.